test/studies/shootout/fannkuch-redux/bradc/fannkuch-redux-blc.chpl

```diff
@@ -1,78 +1,119 @@
 /* The Computer Language Benchmarks Game
- * http://benchmarksgame.alioth.debian.org/
- *
- * contributed by Kyle Brady and Brad Chamberlain
- * based on C implementation by Ledrug Katz
- *
- */
+   http://benchmarksgame.alioth.debian.org/
 
-config const N = 7;
+   contributed by Ben Harshbarger and Brad Chamberlain
+   derived from the Swift implementation by Ralph Ganszky
+*/
 
-assert(N >=3, "N must be in 3..");
+use DynamicIters;
+
+config const n = 7,          // the array size over which to compute perms
+             nchunks = 720;  // the number of chunks of parallelism
+
+//
+// memoize n! (n factorial)
+//
+var fact: [0..n] int;
+fact[0] = 1;
+for i in 1..n do
+  fact[i] = i*fact[i-1];
```

Comment:
I was pretty unhappy with this, but couldn't find a warning-free way to do it that night.  I figured out a way to make it work in a separate PR ( #4101) which I think you'll like better.

(I wanted it out of main() because it seemed more about setting up fact[] than it did about the "main" computation).